### PR TITLE
cleanup(virtctl): More robust VM decoding and minor cleanups in create vm unit and func tests

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does

Make VM decoding a little more robust by using runtime.Decode and apply
some minor cleanups throughout the test files.

In the func tests use the instancetype and preference builders to
construct resources and drop the no longer required workaround in
createVMWithRWOVolume. Also use testsuite.GetTestNamespace(nil) where
appropriate.

To save resources calls to Create now pass the DryRunAll option, because
this suffices to run VMs through the mutating and validating webhooks.

Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Special notes for your reviewer

Merge after #12754

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by alicefr** [comment]
/approve
Nice the dry run option!

**Review by jcanocan — COMMENTED** [review]
Nice job! Just one question: Any chance to extract `runCmd` and `decodeVM` functions to no duplicate them?

File: pkg/virtctl/create/vm/vm_test.go [hunk 1]
```diff
@@ -1088,14 +1095,21 @@ func setFlag(flag, parameter string) string {
 }
 
 func runCmd(args ...string) ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (0xFelix) [reply pkg/virtctl/create/vm/vm_test.go:1097]: I like it short, would keep as is. The package this func is in should give enough context.

File: tests/virtctl/create_vm.go [hunk 2]
```diff
@@ -476,48 +496,41 @@ func setFlag(flag, parameter string) string {
 	return fmt.Sprintf("--%s=%s", flag, parameter)
 }
 
-func unmarshalVM(bytes []byte) *v1.VirtualMachine {
-	vm := &v1.VirtualMachine{}
-	Expect(yaml.Unmarshal(bytes, vm)).To(Succeed())
-	Expect(vm.Kind).To(Equal("VirtualMachine"))
-	Expect(vm.APIVersion).To(Equal("kubevirt.io/v1"))
-	return vm
+func runCmd(args ...string) ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (0xFelix) [reply tests/virtctl/create_vm.go:499]: Same answer :)

**Comment by 0xFelix** [comment]
@jcanocan The funcs are still pretty simple and I don't want to introduce a dependency between unit and func tests.

**Comment by jcanocan** [comment]
> @jcanocan The funcs are still pretty simple and I don't want to introduce a dependency between unit and func tests.

Makes sense.
/lgtm

**Pull request merged**